typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigationInterface.kt

```diff
@@ -3,54 +3,137 @@ package com.xmartlabs.typednavigation
 import androidx.navigation.NamedNavArgument
 import androidx.navigation.NavType
 
-
+/** An interface that implements the basic attributes needed for navigation
+ *
+ * @property name the name of the destination screen
+ * @property url the url string for the navigation
+ * @property navArguments the list of [NamedNavArgument] for the destination screen
+ *
+ */
 sealed interface TypedNavigationInterface {
     val name: String
     val url: String
     val navArguments: List<NamedNavArgument>
 }
 
-interface TypedNavigationInterface0 : TypedNavigationInterface{
+/** An interface that implements navigation without parameters
+ *
+ */
+interface TypedNavigationInterface0 : TypedNavigationInterface {
 
-    fun route() : String
+    /** Get the [String] route for the destination
+     * @return the [String] containing the route of the destination
+     */
```

Comment:
I'd use just the shorter way, but maybe it's not needed
```suggestion
    /** Returns the route of the destination */
```